src/probability/probability_mass_function/basic.lean

```diff
@@ -40,6 +40,9 @@ instance : has_coe_to_fun (pmf α) (λ p, α → ℝ≥0∞) := ⟨λ p a, p.1 a
 @[ext] protected lemma ext : ∀ {p q : pmf α}, (∀ a, p a = q a) → p = q
 | ⟨f, hf⟩ ⟨g, hg⟩ eq :=  subtype.eq $ funext eq
 
+lemma ext_iff (p q : pmf α) : p = q ↔ ∀ x, p x = q x :=
+⟨λ h x, congr_fun (congr_arg _ h) x, pmf.ext⟩
```

Comment:
Could you get there by defining `fun_like (pmf α) (λ _,  α → ℝ≥0∞)` and using `fun_like.ext_iff`?